pkg/envoy/eds/response.go

```diff
@@ -46,7 +49,19 @@ func getEndpointsForProxy(meshCatalog catalog.MeshCataloger, proxyIdentity ident
 			log.Error().Err(err).Msgf("Failed listing allowed endpoints for service %s for proxy identity %s", dstSvc, proxyIdentity)
 			continue
 		}
-		allowedServicesEndpoints[dstSvc] = endpoints
+		var availableEndpoints []endpoint.Endpoint
+		for _, endpoint := range endpoints {
```

Comment:
though I like the use case of this patch a lot, i'm very much worried about this. We are turning several N^2 or worse (from the previous APIs) to to yet another N walk per item (endpoint) on _the entire space of endpoints in the mesh_.

I don't think it can be addressed in the current patch (nor I will ask to), but scalability will not be sustainable if we keep implementing things this way.

Reply:
Yeah, this definitely isn't performant enough yet. This comment of yours went through my head several times: https://github.com/openservicemesh/osm/issues/3105#issuecomment-817084945.

I don't think this change is worth making things significantly slower for now. FWIW in the few times I ran the demo with these changes, I didn't see any instances where this prevented an endpoint from being sent because its proxy was unconnected.

Reply:
That's a good optimization

Reply:
This is a great filter function.  At some point I'd even carve it out in a separate function `filterConnectedEndponits()` or `getAvailableEndpoints()` or something like that.